### PR TITLE
Cookbook added

### PR DESCRIPTION
**Pull request by rahuljain1311**
added cookbook Examples

links added

### CODE REVIEW

**Comment by hueniverse** [comment]
Thanks @rahuljain1311.

@hapijs/support @hapijs/community @hapijs/core-project-maintainers 
Can someone review this and help with making sure the code style is consistent with our rules as well as that this is actually helpful.

File: README.md [hunk 1]
```diff
@@ -118,8 +124,269 @@ emitterObject.emit('test', 2);
...
+const listener1 = function(data) {
+    console.log('listener1 called',data);
+};
+
+podiumObject.on({
+    name: 'event1',
+    count: 2
+},listener1);
+
+podiumObject.emit( 'event1','emit 1');
```
Comment (cjihrig) [inline README.md]: No space after `(`, but add a space after `,`. Applies to other lines.

File: README.md [hunk 2]
```diff
@@ -118,8 +124,269 @@ emitterObject.emit('test', 2);
...
+podiumObject.registerEvent('event1');
+
+const listener1 = function(data) {
+    console.log('listener1 called',data);
+};
+
+podiumObject.on({
+    name: 'event1',
+    count: 2
+},listener1);
```
Comment (cjihrig) [inline README.md]: Space after the comma on this line, and throughout.

File: README.md [hunk 3]
```diff
@@ -118,8 +124,269 @@ emitterObject.emit('test', 2);
...
+```
+### <a name="count"></a>`count`
+```
+const Podium = require('podium');
+const podiumObject = new Podium();
+
+podiumObject.registerEvent('event1');
+
+const listener1 = function(data) {
+    console.log('listener1 called',data);
```
Comment (cjihrig) [inline README.md]: New scopes should begin with a blank line.

File: README.md [hunk 4]
```diff
@@ -118,8 +124,269 @@ emitterObject.emit('test', 2);
...
+    name: 'event1',
+    channels: ['ch1']
+},listener2);
+
+var arr= [0,1,2,3,4,4,5];
+podiumObject.emit({
+    name: 'event1',
+    channel: 'ch1'
+},arr,function(err){
+    if(err){
```
Comment (cjihrig) [inline README.md]: Spacing: `if (err) {`

File: README.md [hunk 5]
```diff
@@ -118,8 +124,269 @@ emitterObject.emit('test', 2);
...
+        console.log("callback 1 error");
```
Comment (cjihrig) [inline README.md]: Should use single quotes.

File: README.md [hunk 6]
```diff
@@ -118,8 +124,269 @@ emitterObject.emit('test', 2);
...
+const Podium = require('podium');
+const podiumObject = new Podium();
+
+podiumObject.registerEvent( [
+    {
+        name: 'event1',
+        channels: ['ch1','ch2'],
+    }
+]);
+podiumObject.registerEvent( [
```
Comment (cjihrig) [inline README.md]: No space after `(`

**Comment by hueniverse** [comment]
This is so close...

**Comment by rahuljain1311** [comment]
@hueniverse weekend is here! What can be improved?

**Comment by hueniverse** [comment]
@rahuljain1311 you have comments from other people that you did not fix yet.

**Comment by AdriVanHoudt** [comment]
cc @devinivy since cookbooks ^^

**Comment by rahuljain1311** [comment]
@hueniverse have solved all the style issues

**Comment by hueniverse** [comment]
I'll let @AdriVanHoudt take over from here.

**Comment by AdriVanHoudt** [comment]
@hueniverse are you ok with the cookbook being in the readme?
@rahuljain1311 I'll take a look when I get home

**Comment by hueniverse** [comment]
I don't really care.

**Comment by hueniverse** [comment]
Should this be merged? @AdriVanHoudt you can merge it or close it.

File: README.md [hunk 7]
```diff
@@ -121,20 +128,300 @@ const source2Object = new Podium('test');
...
-emitterObject.on('test',listener1); // listener1 gets registered on emitterObject, source1Object,source2Object events
-source1Object.on('test',listener2); // listener2 gets registered on source1Object events only
+emitterObject.on('test', listener1); // listener1 gets registered on emitterObject, source1Object,source2Object events
+source1Object.on('test', listener2); // listener2 gets registered on source1Object events only
 
 source1Object.emit('test', 1); // runs all registered events
 emitterObject.emit('test', 2);
 ```
 
+#CookBook - Usage
```
Comment (AdriVanHoudt) [inline README.md]: * Needs a space after the # or github won't render it as a title
* I think it is just `Cookbook` but english is not my native language as well
* I would drop the `Usage` and maybe make subtitle from it

File: README.md [hunk 8]
```diff
@@ -121,20 +128,300 @@ const source2Object = new Podium('test');
...
+
+### <a name="channels"></a>`channels`
+```
```
Comment (AdriVanHoudt) [inline README.md]: Make these ```js, github will syntax highlight the code which makes it way more readable

File: README.md [hunk 9]
```diff
@@ -121,20 +128,300 @@ const source2Object = new Podium('test');
...
+    {
+        name: 'event1',
+        channels: ['ch1', 'ch2', 'ch3', 'ch4'],
+    },
+    {
+        name: 'event2',
+        channels: ['ch1', 'ch2']
+    }
+]);
+const listener1 = (data) => { //using ES6
```
Comment (AdriVanHoudt) [inline README.md]: only es6 is fine, unless `this` has a meaning?

File: README.md [hunk 10]
```diff
@@ -121,20 +128,300 @@ const source2Object = new Podium('test');
...
+    channels: ['ch3', 'ch4']
+}, listener2);
+
+
+podiumObject.on({ name: 'event1', channels: 'ch2' }, (data) => { // autonomous function
+
+    console.log('auto', data);
+});
+
+var arr= [0, 1, 2, 3, 4, 4, 5];
```
Comment (AdriVanHoudt) [inline README.md]: space after `arr`

File: README.md [hunk 11]
```diff
@@ -121,20 +128,300 @@ const source2Object = new Podium('test');
...
+    if (err){
+        console.log('callback 2 error');
+    }
+    else{
+        console.log('callback 2 returned true!');
+    }
+});
+console.log('after event2, ch1: ', arr);
+```
+###<a name="spread"></a>`spread`
```
Comment (AdriVanHoudt) [inline README.md]: needs a space after the ###

File: README.md [hunk 12]
```diff
@@ -121,20 +128,300 @@ const source2Object = new Podium('test');
...
+        channels: ['ch1', 'ch2'],
+        spread: true
+    },
+    {
+        name: 'event2',
+        channels: ['ch1', 'ch2']
+    }
+]);
+
+const listener1 =(data1, data2, data3, data4) => { //using ES6
```
Comment (AdriVanHoudt) [inline README.md]: space after `=`

File: README.md [hunk 13]
```diff
@@ -121,20 +128,300 @@ const source2Object = new Podium('test');
...
+    name: 'event2',
+    channels: ['ch1']
+}, listener2);
+
+var arr= [0, 1, 2, 3, 4, 4, 5];
+console.log('intially: ', arr);
+podiumObject.emit({
+    name: 'event1',
+    channel: 'ch1'
+},arr,function(err){
```
Comment (AdriVanHoudt) [inline README.md]: spaces after ,

File: README.md [hunk 14]
```diff
@@ -121,20 +128,300 @@ const source2Object = new Podium('test');
...
+    if (err){
+        console.log('callback 2 error');
+    }
+    else{
+        console.log('callback 2 returned true!');
+    }
+});
+console.log('after event2, ch1: ', arr);
+```
+###<a name="shared"></a>`shared`
```
Comment (AdriVanHoudt) [inline README.md]: space after ###

File: README.md [hunk 15]
```diff
@@ -38,29 +37,37 @@ podiumObject.registerEvent({
  });
 ```
 
+- [`channels`](#channels)
```
Comment (AdriVanHoudt) [inline README.md]: Add some description that these links refer to examples for all the options

File: README.md [hunk 16]
```diff
@@ -38,29 +37,37 @@ podiumObject.registerEvent({
...
+
 ## `podium.on(criteria, listener)`
 
 Subscribe a handler to an event. Handler can be seen a function which will be called when the event occurs.
 
 ```javascript
 podiumObject.registerEvent('event1');
+podiumObject.on('event1', function(update){ // Way 1
 
-podiumObject.on('event1',function(update){ // Way 1
```
Comment (AdriVanHoudt) [inline README.md]: why remove the space?

File: README.md [hunk 17]
```diff
@@ -38,29 +37,37 @@ podiumObject.registerEvent({
...
     console.log('inside autonomous listener without name! data:', update);
 });
 
-const listener1 = function() { // normal function object
+const listener1 = function(){ // normal function object
```
Comment (AdriVanHoudt) [inline README.md]: Why remove the space?

**Comment by rahuljain1311** [comment]
@AdriVanHoudt have solved the comments!

**Comment by AdriVanHoudt** [comment]
@rahuljain1311 👏 thanks for the contribution! Really awesome work 🎉

**Pull request merged**